File: BaekJoon/15652.cpp
#include <iostream>
#include <string>
#include <cstring>
#include <cctype>
#include <vector>
#include <algorithm>

using namespace std;

int n, m, cnt;
int arr[9] = { 0, };

void solution(int start, int cnt) {
	// 비내림차순(같은 수 여러번 골라도 됨)
	if (cnt == m) {
		for (int j = 0; j < m; j++) {
			cout << arr[j] << ' ';
		}
		cout << '\n';
		return;
	}

	for (int i = start; i <= n; i++) {
		arr[cnt] = i;
		solution(i, cnt + 1);
	}
}


void input() {
	cin >> n >> m;
}

int main() {
	input();
	solution(1, 0);
}

File: BaekJoon/15652.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())

result = []
def sol(start):
  if len(result) == m:
    print(*result)
    return

  # 비내림차순으로 들어가야함
  for i in range(start, n + 1):
    result.append(i)
    sol(i)
    result.pop()

sol(1)

#####################
# itertools 쓰는 방법 (이런 함수도 있었군,,)

import sys
from itertools import combinations_with_replacement as combwr

n, m = map(int, input().split())
arr = range(1, n + 1)
for i in combwr(arr, m):
  print(*i)
